cachito/web/content_manifest.py

```diff
@@ -71,8 +73,25 @@ def process_gomod(self, package, dependency):
         :param Dependency dependency: the gomod package dependency to process
         """
         if dependency.type == "gomod":
-            parent_purl = self._gomod_data[package.name]["purl"]
-            dep_purl = to_purl(dependency)
+
+            parent_module_name = package.name
+            relpath_from_parent_module_to_dep = None
+
+            # if we're processing a local dependency
+            if dependency.version and dependency.version.startswith("."):
+                dep_normpath = os.path.normpath(os.path.join(package.name, dependency.version))
+                parent_module_name = gomod.match_parent_module(
+                    dep_normpath, self._gomod_data.keys()
+                )
+                # if the dependency is not in this repo, use a relative path from the root module
+                # otherwise just use the parent purl directly
+                if dependency.name not in self._gomod_data:
+                    relpath_from_parent_module_to_dep = Path(dep_normpath).relative_to(
+                        Path(parent_module_name)
+                    )
```

Comment:
It took me a while to understand why `dependency.name in self._gomod_data` is a special case. For others: it would mean that this dependency is one of the modules that the user specified in the Request. We processed the module => we have a regular purl for it, we don't need to use a subpath-based purl.

~This seems correct, but I wonder if we need to special-case it. If the dependency is in gomod_data, then `gomod.match_parent_module(dep_normpath, gomod_data) == dep_normpath`. The dependency is its own parent module, the path from parent to dependency is an empty `Path()`.~

~(we would instead have logic for empty paths in `to_purl()`)~

Edit: scratch this, we need this special case for go-packages